### PR TITLE
Add option to lock power device while printing

### PR DESCRIPTION
**Pull request by jordanruthe**
This adds an option to lock a power device while printing. Specifically, it ensures that you cannot turn off the printer while the printer is in a printing state.

### CODE REVIEW

File: moonraker/plugins/power.py [hunk 1]
```diff
@@ -97,6 +104,14 @@ async def _handle_power_request(self, web_request):
...
     async def _process_request(self, device, req):
         if req in ["on", "off"]:
+            printing = await self._check_klippy_printing()
+            logging.info("Device locked: %s" % device.get_locked_while_printing())
+            logging.info("Klippy printing: %s" % printing)
+            if device.get_locked_while_printing() and printing:
+                self.server.error(f"Unable to change power for {device} " +
+                    "while printing")
+                dev_info = device.get_device_info()
+                return dev_info['status']
```
Comment (Arksine) [inline moonraker/plugins/power.py:114]: I think you need to raise the exception.  When doing so it won't be necessary to return device status, the client will receive the error instead.

**Review by Arksine — COMMENTED** [review]
Thanks!  Overall the changes look good to me. We just need to raise the exception when a power request is made while printing.

File: moonraker/plugins/power.py (hunk 1 above)
Comment (jordanruthe) [reply moonraker/plugins/power.py]: Thanks for the feedback. I just changed it to raise an error

**Comment by Arksine** [comment]
Thanks!

**Pull request merged**